Migrate/unit details editor

sammyrolfe opened the pull request:
_Any italic text should be deleted from the final Pull Request text, including this line_

# Description

Migrate the unit-details-editor component from CoffeeScript and Angular.js to TypeScript and Angular. The component is used to update details and settings for a particular unit.

![Unit Details Editor Before](https://github.com/thoth-tech/doubtfire-web/assets/36840091/56842783-9983-446d-a9dc-8e6ce48f15a7)



![Unit Details Editor After](https://github.com/thoth-tech/doubtfire-web/assets/36840091/0e84c02a-581e-4259-bd2d-4aa9eaf7ea2c)


## Type of change


- [ ] Migration of component to Angular17 and Typescript

# How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration_

## Testing Checklist:

- [ ] Tested in latest Chrome
- [ ] Tested in latest Safari
- [ ] Tested in latest Firefox

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have requested a review from @macite and @jakerenzella on the Pull Request

sammyrolfe committed:
Created initial files for migration

sammyrolfe committed:
feat: component-functional

sammyrolfe committed:
ready for review

sammyrolfe committed:
added draft learning summary help block

sammyrolfe commented:
Update - Added missing hint below draft learning summary

prabhjotcodes commented:
PR #201  - Review
Dated - 16/05/24
By - Prabhjot Singh ( sID: 223074182 )

Following issues were found while reviewing `unit-details-editor` component:

\- Draft Learning Summary, Select Box is not selectable
\- “Disabled” Text has to be removed next to the toggle component, which is not present in the existing component, Even if it is present on toggling ON, the text didn’t change to “Enabled”
\- Extension duration on resubmit, Text Input should be turned into Angular material text input
\- The “Reorder” Button should be moved to the left and the “Update” Button should be moved to the right in the footer section
\- In unit-details-editor.component.html, bootstrap buttons have to be changed to angular material buttons, with its hover effect
\- A little more space to be added between “Update Unit” or “Create Unit” and “Create a new unit with all overview unit details here.” or “Update overview details of the unit below in the header section, similar to the existing component

The above issues can be resolved by replacing bootstrap elements with angular material components, removing elements that are not in the existing component, Fixing Spacing, and following the order of placement of elements like how it is in the existing component.

sammyrolfe committed:
revision 1

maddernd commented:
> In unit-details-editor.component.html, bootstrap buttons have to be changed to angular material buttons, with its hover effect
> - A little more space to be added between “Update Unit” or “Create Unit” and “Create a new unit with all overview unit details here.” or “Update overview details of the unit below in the header section, similar to the existing component

This is a pretty good review, good job on that, these points all need addressing.

sammyrolfe committed:
revision-2

sammyrolfe commented:
Hi @PrabhKamboj, thanks for your feedback. I forgot to include the edit.tpl file where my component was loaded so it wasn't passing the unit object through correctly. This has now been resolved and I've updated as per your comments.
Thanks again, appreciate your feedback.

prabhjotcodes reviewed (changes requested):
PR #201 - Revised Review
Dated - 22/05/24
By - Prabhjot Singh ( sID: 223074182 )

From my first review of this PR following issues are fixed,
\- Draft Learning Summary, Select Box is not selectable
\- “Disabled” Text has to be removed next to the toggle component, which is not present in the existing component, Even if it is present on toggling ON, the text didn’t change to “Enabled”
\- Extension duration on resubmit, Text Input should be turned into Angular material text input
\- The “Reorder” Button should be moved to the left and the “Update” Button should be moved to the right in the footer section
\- In unit-details-editor.component.html, bootstrap buttons have to be changed to angular material buttons, with its hover effect
\- A little more space to be added between “Update Unit” or “Create Unit” and “Create a new unit with all overview unit details here.” or “Update overview details of the unit below in the header section, similar to the existing component

**However**, After reviewing the latest changes, I have some feedback below:
\- Now Both the button in the footer is named “Rollover” which is supposed to be “Rollover” and next to it, it should be “**Update**”
\- The button's Background color is blue, But as per the existing component, it should be in a green shade.

Other than the new feedback, the rest of the issues are fixed from the first review.

sammyrolfe committed:
revision-3

sammyrolfe commented:
Hi @PrabhKamboj,

Thanks for that feedback, I've now adjusted the buttons to use the correct colour and fixed the naming on the update button.

Thanks
Sam

prabhjotcodes reviewed (approved):
PR #201 - Finalized Review
Dated - 22/05/24
By - Prabhjot Singh ( sID: 223074182 )

From the last review I made, the following issues have been fixed,
\- Now Both the button in the footer is named “Rollover” which is supposed to be “Rollover” and next to it, it should be “Update”
\- The button's Background color is blue, But as per the existing component, it should be in a green shade

The student managed to fix all the issues and the system is now working fine same as the existing component. This pull request is completed and it's ready to be merged.
